src/utils/permalinks/Permalinks.js

```diff
@@ -130,7 +130,13 @@ export class RoomPermalinkCreator {
     }
 
     forRoom() {
-        return getPermalinkConstructor().forRoom(this._roomId, this._serverCandidates);
+        try {
+            // Prefer to use canonical alias for permalink if possible
+            const alias = this._room.getCanonicalAlias();
+            return getPermalinkConstructor().forRoom(alias, this._serverCandidates);
+        } catch (error) {
```

Comment:
we don't use the try/catch methodology for these things - we instead use if statements to check truthiness of the alias itself.

Reply:
Thanks for the feedback. I was more so concerned with both `this._room` being a valid instance of a Room which I could call `getCanonicalAlias()` on. Have pushed a new patch.

I have followed the pattern used here [https://github.com/matrix-org/matrix-react-sdk/blob/cf8c98e07665ec8f68846a5524b566d980c22e90/src/utils/permalinks/Permalinks.js#L80-L80] to check the validity of the room and therefore `this._room` reference.

This check appears to presume that room will either be a valid room object or it will be `null` or `undefined`.